backend/de.metas.contracts/src/main/java/de/metas/contracts/finalinvoice/invoicecandidate/FlatrateTermModular_Handler.java

```diff
@@ -0,0 +1,387 @@
+import de.metas.invoicecandidate.model.X_C_Invoice_Candidate;
+import de.metas.invoicecandidate.spi.IInvoiceCandidateHandler;
+import de.metas.lang.SOTrx;
+import de.metas.lock.api.LockOwner;
+import de.metas.money.CurrencyId;
+import de.metas.organization.OrgId;
+import de.metas.pricing.PricingSystemId;
+import de.metas.product.ProductPrice;
+import de.metas.quantity.Quantity;
+import de.metas.quantity.Quantitys;
+import de.metas.tax.api.ITaxDAO;
+import de.metas.tax.api.Tax;
+import de.metas.tax.api.TaxCategoryId;
+import de.metas.tax.api.TaxId;
+import de.metas.tax.api.TaxNotFoundException;
+import de.metas.tax.api.TaxQuery;
+import de.metas.uom.UomId;
+import de.metas.util.Check;
+import de.metas.util.Services;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import org.adempiere.ad.dao.QueryLimit;
+import org.adempiere.ad.trx.api.ITrxManager;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.compiere.SpringContextHolder;
+import org.compiere.model.X_C_DocType;
+
+import java.math.BigDecimal;
+import java.util.Iterator;
+import java.util.List;
+
+import static java.util.Collections.emptyIterator;
+
+public class FlatrateTermModular_Handler implements ConditionTypeSpecificInvoiceCandidateHandler
+{
+	private final IDocTypeBL docTypeBL = Services.get(IDocTypeBL.class);
+	private final ITaxDAO taxDAO = Services.get(ITaxDAO.class);
+	private final IInvoiceCandBL invoiceCandBL = Services.get(IInvoiceCandBL.class);
+	private final ITrxManager trxManager = Services.get(ITrxManager.class);
+
+	private final ModularContractSettingsDAO modularContractSettingsDAO = SpringContextHolder.instance.getBean(ModularContractSettingsDAO.class);
+	private final ModularContractLogDAO modularContractLogDAO = SpringContextHolder.instance.getBean(ModularContractLogDAO.class);
+	private final ModularContractService modularContractService = SpringContextHolder.instance.getBean(ModularContractService.class);
+	private final ModularContractComputingMethodHandlerRegistry modularContractComputingMethods = SpringContextHolder.instance.getBean(ModularContractComputingMethodHandlerRegistry.class);
+
+	@Override
+	public String getConditionsType()
+	{
+		return X_C_Flatrate_Term.TYPE_CONDITIONS_ModularContract;
+	}
+
+	/**
+	 * @return empty iterator
+	 */
+	@Override
+	public Iterator<I_C_Flatrate_Term> retrieveTermsWithMissingCandidates(@NonNull final QueryLimit limit)
+	{
+		return emptyIterator();
+	}
+
+	@Override
+	public List<I_C_Invoice_Candidate> createInvoiceCandidates(
+			@NonNull final I_C_Flatrate_Term term,
+			@NonNull final LockOwner lockOwner)
+	{
+		return createModularICs(term, lockOwner);
+	}
+
+	@Override
+	public Quantity calculateQtyEntered(final I_C_Invoice_Candidate invoiceCandidateRecord)
+	{
+		final UomId uomId = HandlerTools.retrieveUomId(invoiceCandidateRecord);
+
+		return Quantitys.create(invoiceCandidateRecord.getQtyEntered(), uomId);
```

Comment:
why are we taking the qty and uom from 2 different places?!

same question for de.metas.contracts.modular.interim.invoice.invoicecandidatehandler.FlatrateTermInterimInvoice_Handler#calculateQtyEntered too .

Reply:
FlatrateTermInterimInvoice_Handler will be handled in its issue